lib/aarch64/misc_helpers.S

```diff
@@ -24,9 +24,9 @@
 	.globl	disable_mmu_el3
 	.globl	disable_mmu_icache_el1
 	.globl	disable_mmu_icache_el3
-
+#if ENABLE_PIE
```

Comment:
@soby-mathew 

Replied above

Reply:
As commented, unused function definitions are meant to be removed by the linker. Once that is enabled, this will not be required. It is not scalable/maintainable to wrap function definition for every build option they are meant to compile with. We expect the linker to do the job.

Reply:
@soby-mathew 

Even if we enable garbage collection including code under proper macros helps readability and makes the intention explicit. I believe this makes the code easy to read/understand. I still prefer getting this patch merged.

Reply:
I think the comments are sufficient for making the intention/readability of the function. The overall intention in the project is to reduce conditional compilation within source files when possible.

Reply:
@soby-mathew 

I disagree. The code accesses linker variables that are not available if ENABLE_PIE is disabled. Removing the checks from this patch leads to the following errors

'Error: L6218E: Undefined symbol __GOT_END__ (referred from misc_helpers.o).
Error: L6218E: Undefined symbol __GOT_START__ (referred from misc_helpers.o).
Error: L6218E: Undefined symbol __RELA_END__ (referred from misc_helpers.o).
Error: L6218E: Undefined symbol __RELA_START__ (referred from misc_helpers.o).'